WireAPI/Sources/WireAPI/Private/APIs/BackendInfoAPI/BackendInfoAPIV1.swift

```diff
@@ -0,0 +1,25 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class BackendInfoAPIV1: BackendInfoAPIV0 {
```

Comment:
I don't understand why we have this chain of inheritance of api versions, can not each version just conform to the protocol `BackendInfoAPI`?
This would make it easier to understand where a certain logic is implemented and also remove versions in the future.

Reply:
This is one approach that I implemented that is the same that Android does. I'll also implement another approach without inheritance, so we can compare what we prefer.

Reply:
This was debated at length in the Android team. 

**Inheritance**
+ Minimum code duplication
+ It's clear from reading the code what changed with each API version
- Deleting old API versions require some care a code restructuring 

**Without  Inheritance**
+ No dependencies so very simple to delete old API versions
- A lot of code duplication, so any fixes must be applied in several places.

Reply:
thanks for the summary

Reply:
They use inheritance